src/main/java/bio/terra/service/auth/iam/sam/SamIam.java

```diff
@@ -85,8 +85,14 @@ private ApiClient getApiClient(String accessToken) {
     ApiClient apiClient = new ApiClient();
     apiClient.setAccessToken(accessToken);
     apiClient.setUserAgent("OpenAPI-Generator/1.0.0 java"); // only logs an error in sam
-    apiClient.setConnectTimeout(
-        configurationService.getParameterValue(ConfigEnum.SAM_OPERATION_TIMEOUT_SECONDS));
+
+    // Sometimes Sam calls can take longer than the OkHttp default of 10 seconds to return a
+    // response.  In those cases, we can see socket timeout exceptions despite the underlying Sam
+    // call continuing to execute and possibly succeeding.
+    int operationTimeoutSeconds =
+        configurationService.getParameterValue(ConfigEnum.SAM_OPERATION_TIMEOUT_SECONDS);
```

Comment:
Is there a reason to not increase the config value here?

Reply:
No, I don't think that's necessary.  We don't really want to leave TDR's caller hanging for much longer than 60 seconds (that's the value of `SAM_OPERATION_TIMEOUT_SECONDS`). This PR gives more flexibility as to how those 60 seconds are spent: if our calls to Sam fail fast, then we'll be able to wait and try again 1-2 more times.  But if our calls to Sam need more time, we'll take that time, up to 60 seconds.

Reply:
This code was using seconds already, so I think it's OK.

Reply:
Ah, I missed that the difference was seconds -> millis. This looks good.